ospfd/ospf_ext.h

```diff
@@ -167,18 +187,15 @@ struct ext_itf {
 	/* Area info in which this SR link belongs to. */
 	struct ospf_area *area;
 
-	/* Flags to manage this link parameters. */
-	uint32_t flags;
-
 	/* SID type: Node, Adjacency or LAN Adjacency */
 	enum sid_type stype;
 
-	/* extended link/prefix TLV information */
-	struct ext_tlv_prefix prefix;
-	struct ext_subtlv_prefix_sid node_sid;
-	struct ext_tlv_link link;
-	struct ext_subtlv_adj_sid adj_sid[2];
-	struct ext_subtlv_lan_adj_sid lan_sid[2];
+	union {
+		/* list of extended prefix TLV information */
+		struct list *prefix_sid_list;
+		/* extended link TLV information */
+		struct link_lsa link_lsa;
+	} lsa;
```

Comment:
I understand the list for the prefix SID, but why adding also a list for link LSA ? As per RFC7684, it is mention that Opaque Extended Link could convey only one Link TLV per Opaque LSA:

> 3.1.  OSPFv2 Extended Link TLV

>  The OSPFv2 Extended Link TLV is used to advertise various attributes
   of the link.  It describes a single link and is constructed of a set
   of sub-TLVs.  There are no ordering requirements for the sub-TLVs.
   Only one OSPFv2 Extended Link TLV SHALL be advertised in each OSPFv2
   Extended Link Opaque LSA, allowing for fine granularity changes in
   the topology.